ui/app/controllers/jobs/run/templates/index.js

```diff
@@ -1,17 +1,30 @@
 import Controller from '@ember/controller';
 import { action } from '@ember/object';
 import { tracked } from '@glimmer/tracking';
+import { capitalize } from '@ember/string';
 
 export default class JobsRunTemplatesController extends Controller {
   @tracked selectedTemplate = null;
 
   get templates() {
     return this.model.map((templateVariable) => {
       // THIS LOGIC SHOULD LIKELY MOVE TO THE SERIALIZATION LAYER
-      const description = templateVariable.items.description;
+      const description = templateVariable.keyValues.findBy('key', 'description')?.value;
+
+      // Removes the preceeding nomad/job-templates/default/, as well as the namespace, from the ID
+      let label;
+      const delimiter = templateVariable.id.lastIndexOf('/');
+      if (delimiter !== -1) {
+        label = templateVariable.id.slice(delimiter+1);
+      } else {
+        label = templateVariable.id
+      }
+
+      label = capitalize(label.split('@')[0].replace(/-/g, ' '));
+  
       return {
         id: templateVariable.id,
-        label: templateVariable.path.split('nomad/job-templates/')[1],
```

Comment:
Moving from .path to .id here because the default templates are not Variables, but just Plain Old Javascript Objects.

Reply:
ditto.